src/main/java/org/java_websocket/server/WebSocketServer.java

```diff
@@ -1079,8 +1079,20 @@ public void run() {
         }
       } catch (InterruptedException e) {
         Thread.currentThread().interrupt();
-      } catch (RuntimeException e) {
-        handleFatal(ws, e);
+      } catch (VirtualMachineError | ThreadDeath | LinkageError e) {
+        if (ws != null) {
+          ws.close();
+        }
+        log.error("Got fatal error in worker thread" + getName());
```

Comment:
This could use logger parameter `{}` instead of string concat

Reply:
fixed